### PR TITLE
Cli epoch-info: generate epoch-completed time from block times

### LINKED ISSUE #17472: solana epoch-info mis-estimates epoch duration as epoch end nears
#### Problem

solana epoch-info is mis-estimating the current epoch duration, and it's getting less and less accurate as the end of epoch nears.  This is epoch 184 on mainnet-beta.  The epoch is lasting approximately 3 days 4 hours, and the solana epoch-info command roughly showed that estimate up to a couple of hours ago, but as the end of epoch approaches, the display has been getting less and less accurate.

$ solana epoch-info

Block height: 72629917
Slot: 79918511
Epoch: 184
Transaction Count: 18232557494
Epoch Slot Range: [79488000..79920000)
Epoch Completed Percent: 99.655%
Epoch Completed Slots: 430511/432000 (1489 remaining)
Epoch Completed Time: 2days 21h 28m 46s/2days 21h 43m 12s (14m 25s remaining)

/sandbox/b-hydra-mainline/home-working/srcroot$ solana block-time 79488000
Block: 79488000
Date: 2021-05-22T03:07:36Z

$ date -u
Tue 25 May 2021 06:58:09 AM UTC

As you can see, the epoch began 3 days 3 hours 51 minutes ago.  And it has approximately 14 minutes remaining.  But solana epoch-info is now showing an epoch completed time of 2 days 21 hours 28 minutes, with only 14 minutes left.

### PR DESCRIPTION
**Pull request by CriesofCarrots**
#### Problem
`solana epoch-info` uses recent performance samples to estimate the epoch's elapsed and total times. However, the samples only cover the last hour. This is a good enough estimate for the remaining epoch time, but we actually know the elapsed epoch time from the block times.

#### Summary of Changes
Use block times for the epoch-start and current Slots to calculate epoch-completed time and refine the total epoch time estimate.

Fixes #17472

### CODE REVIEW

File: cli-output/src/cli_output.rs [hunk 1]
```diff
@@ -277,21 +281,29 @@ impl fmt::Display for CliEpochInfo {
                 remaining_slots_in_epoch
             ),
         )?;
+        let time_elapsed = if let (Some(start_block_time), Some(current_block_time)) =
+            (self.start_block_time, self.current_block_time)
+        {
+            Duration::from_secs((current_block_time - start_block_time) as u64)
+        } else {
```
Comment (t-nelson) [inline cli-output/src/cli_output.rs:288]: Should we annotate the fallback case somehow? Maybe `(imprecise)`, not that either is particularly "precise" :sweat_smile:

Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs:288]: Yeah, some kind of annotation is a good idea. I'll think on a good term....

Comment (CriesofCarrots) [reply cli-output/src/cli_output.rs:291]: @t-nelson what do you think of https://github.com/solana-labs/solana/pull/18258/commits/9e5f96c6bd1adec4da68b1ee6d32d1b73d0c0c7e ? I tried a bunch of one-word tags -- before everything, after everything, or by the affected data -- but didn't find anything that was both clear and non-disruptive to the `Epoch Completed Time:` line (which is already complex, due to parenthetical)

Comment (t-nelson) [reply cli-output/src/cli_output.rs:291]: Works for me!

NIT: Maybe an extra newline in front of the annotation to make it stand out?

**Review by t-nelson — APPROVED** [review]
lgtm!

**Pull request merged**